src/cpp/rtps/builtin/discovery/database/DiscoveryParticipantsAckStatus.hpp

```diff
@@ -45,15 +45,25 @@ class DiscoveryParticipantsAckStatus
 
     ~DiscoveryParticipantsAckStatus() = default;
 
+    enum class ParticipantState : uint8_t
+    {
+        UNMATCHED,  // Data(p) has not been sent yet
+        SENT,       // Data(p) has already been sent
+        MATCHED     // Data(p) has been acked
+    };
+
     void add_or_update_participant(
             const GuidPrefix_t& guid_p,
-            bool status);
+            ParticipantState status);
 
     void remove_participant(
             const GuidPrefix_t& guid_p);
 
     void unmatch_all();
 
+    bool is_sent(
```

Comment:
If the previous comment is accepted, this would be either a `is_pending_send()` or even `is_waiting_ack()`